Data_Structures_Questions.md

```diff
@@ -3,51 +3,56 @@ Answer the following questions for each of the data structures you implemented a
 ## Queue
 
 1. What is the runtime complexity of `enqueue`?
-
+   O(1)
 2. What is the runtime complexity of `dequeue`?
-
+   O(1)
 3. What is the runtime complexity of `len`?
+   O(n)
 
 ## Binary Search Tree
 
-1. What is the runtime complexity of `insert`? 
-
+1. What is the runtime complexity of `insert`?
+   O(log n)
 2. What is the runtime complexity of `contains`?
-
-3. What is the runtime complexity of `get_max`? 
+   O(log n)
+3. What is the runtime complexity of `get_max`?
+   O(log n)
 
 ## Heap
 
 1. What is the runtime complexity of `_bubble_up`?
-
+   O(logn)
 2. What is the runtime complexity of `_sift_down`?
-
+   O(logn)
 3. What is the runtime complexity of `insert`?
-
+   (logn) constant(append) + bubble up
 4. What is the runtime complexity of `delete`?
-
+   O(logn)
 5. What is the runtime complexity of `get_max`?
```

Comment:
Hahaha, and if we have a generic heap thingy, we can also get it to do `get_min` whooo. cool stuff.